lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+class OmiseCapabilities extends OmiseApiResource
+{
+    const ENDPOINT = 'capability';
+
+    /**
+     * Retrieves capabilities.
+     *
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @return OmiseCapabilities
+     */
+    public static function retrieve($publickey = null, $secretkey = null)
+    {
+        return parent::g_retrieve(get_class(), self::getUrl(), $publickey, $secretkey);
+    }
+
+    /**
+     * Retrieves array of payment backends. Optionally pass in as many filter functions as you want
+     *
+     * @param [function,...]
+     *    
+     * @return array
+     */
+    public function getBackends() {
+        // check for filters
+        if ($filters = func_get_args()) $filter = self::combineFilters($filters);
+        $res = $this['payment_backends'];
+        array_walk(
+            $res,
+            function($v, $k) use (&$res) {
+                $id = array_keys($v)[0];
+                $res[$k][$id]['_id'] = $id;
+            }
+        );
+        $res = array_map(function($a) { return (object)reset($a);}, $res);
+        return !empty($filter) ? array_filter($res, $filter) : $res;
+    }
+
+    /**
+     * Makes a filter function to check supported currency for backend.
+     *
+     * @param  string $currency
+     *
+     * @return function
+     */
+    public function backendSupportsCurrency($currency) {
+        return function($backend) use ($currency) { return in_array($currency, $backend->currencies); };
+    }
+
+    /**
+     * Makes a filter function to check type of backend.
+     *
+     * @param  string $type
+     *
+     * @return function
+     */
+    public function backendTypeIs($type) {
+        return function($backend) use ($type) { return $backend->type==$type; };
+    }
+
+    /**
+     * Makes a filter function to check if backends can handle given amount.
+     *
+     * @param  int $amount
+     *
+     * @return function
+     */
+    public function backendSupportsChargeAmount($amount) {
```

Comment:
Also while `backendTypeIs`, `backendSupportsChargeAmount`, `backendSupportsCurrency` sound tempting but it's hard to differentiate between its filter methods and normal method. 
How about we prefix it with `filter`? 

```php
$capability->getBackends(
    $capability->filterBackendType('card'),
    $capability->filterBackendSupportsChargeAmount(3000)
);
```

--
Actually not sure which one would be ideal, it could look like 👇 as well.
```php
$capability->getBackends([
    'filters' => [
        'type'                 => 'card',
        'supportsChargeAmount' => 3000
    ]
]);
```

🤔 hmm, what do you think?

Reply:
Doing it with the functions easily allows people to add their own filters if required without touching the omise-php code or requesting a change (they simply pass in their own filter function). I have no strong opinions on the name change for the filter building methods

Reply:
I've pushed a change with the above naming change

Reply:
should we add `3000` to `const` @jonrandy ?

Reply:
3000 does not appear in the code - the above is an example

Reply:
@guzzilar Actually, what about this?

```php
$capabilities->getBackends(
    $capabilities->backendFilters['type']('card'),
    $capabilities->backendFilters['currency']('THB')
    $capabilities->backendFilters['amount'](3000)
);
```
I think it reads a bit nicer and still allows the user to pass in their own filter functions

Reply:
Not sure what PHP version supports this construction though... checking